client/map_verifier.go

```diff
@@ -86,3 +88,26 @@ func (m *MapVerifier) VerifyMapLeafInclusionHash(rootHash []byte, leafProof *tri
 func (m *MapVerifier) VerifySignedMapRoot(smr *trillian.SignedMapRoot) (*types.MapRootV1, error) {
 	return tcrypto.VerifySignedMapRoot(m.PubKey, m.SigHash, smr)
 }
+
+// VerifyMapLeavesResponse verifies the responses of GetMapLeaves and GetMapLeavesByRevision.
+// To accept any map revision, pass -1 as revision.
+func (m *MapVerifier) VerifyMapLeavesResponse(indexes [][]byte, revision int64, resp *trillian.GetMapLeavesResponse) ([]*trillian.MapLeaf, error) {
+	if got, want := len(resp.MapLeafInclusion), len(indexes); got != want {
+		return nil, status.Errorf(codes.Internal, "got %v leaves, want %v", got, want)
```

Comment:
The other errors start with "VerifySignedMapRoot()".

Reply:
I think only the error from `VerifySignedMapRoot` starts that way...

Reply:
Fair enough but I think we could do with improving some messages throughout the codebase, probably a lot of them belong to me :-)